### PR TITLE
Add hub routing preference in virtual hub, deprecated PreferredRoutingGateway

### PR DESCRIPTION
**Pull request by zaynchen115**
## Description

Add hub routing preference in virtual hub properties to allow customer to select the routing preference of the virtual hub, deprecated previous property PreferredRoutingGateway

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by zaynchen115** [comment]
Enabled required feature in Test Subscription

File: src/Network/Network.Test/ScenarioTests/CortexTests.ps1 [hunk 1]
```diff
@@ -68,11 +68,12 @@ function Test-CortexCRUD
 		Assert-NotNull $virtualWansAll
 
 		# Create the Virtual Hub
-		$createdVirtualHub = New-AzVirtualHub -ResourceGroupName $rgName -Name $virtualHubName -Location $rglocation -AddressPrefix "192.168.1.0/24" -VirtualWan $virtualWan
+		$createdVirtualHub = New-AzVirtualHub -ResourceGroupName $rgName -Name $virtualHubName -Location $rglocation -AddressPrefix "192.168.1.0/24" -VirtualWan $virtualWan -HubRoutingPreference "ASPath"
 		$virtualHub = Get-AzVirtualHub -ResourceGroupName $rgName -Name $virtualHubName
 		Assert-AreEqual $rgName $virtualHub.ResourceGroupName
 		Assert-AreEqual $virtualHubName $virtualHub.Name
 		Assert-AreEqual "192.168.1.0/24" $virtualHub.AddressPrefix
+		Assert-AreEqual "ASPath" $virtualHub.HubRoutingPreference
```
Comment (VeryEarly) [inline src/Network/Network.Test/ScenarioTests/CortexTests.ps1:76]: test failed at this line, please check

**Comment by VeryEarly** [comment]
please target against main branch

**Comment by zaynchen115** [comment]
Test not passes due to the SDK version is not updated

**Comment by isra-fel** [comment]
Kindly ping @zaynchen115 , what's the next step?

**Pull request closed without merging**